Framework/Crystal/src/CalibrationHelpers.cpp

```diff
@@ -0,0 +1,165 @@
+#include "MantidAPI/DetectorInfo.h"
+#include "MantidCrystal/SCDCalibratePanels.h"
+#include "MantidCrystal/CalibrationHelpers.h"
+
+using namespace Mantid::API;
+using namespace Mantid::Geometry;
+using namespace Mantid::Kernel;
+
+namespace Mantid {
+namespace Crystal {
+
+namespace {
+constexpr double RAD_TO_DEG = 180. / M_PI;
+}
+
+/**
+ * Converts a Quaternion to a corresponding matrix produce Rotx*Roty*Rotz,
+ * corresponding to the order Mantid uses in calculating rotations
+ * @param Q The Quaternion. It will be normalized to represent a rotation
+ * @param Rotx The angle in degrees for rotating around the x-axis
+ * @param Roty The angle in degrees for rotating around the y-axis
+ * @param Rotz The angle in degrees for rotating around the z-axis
+ */
+void CalibrationHelpers::quatToRotxRotyRotz(const Quat Q, double &Rotx,
+                                            double &Roty, double &Rotz) {
+  Quat R(Q);
+  R.normalize();
+  V3D X(1, 0, 0);
+  V3D Y(0, 1, 0);
+  V3D Z(0, 0, 1);
+  R.rotate(X);
+  R.rotate(Y);
+  R.rotate(Z);
+  if (Z[1] != 0 || Z[2] != 0) {
+    double tx = atan2(-Z[1], Z[2]);
+    double tz = atan2(-Y[0], X[0]);
+    double cosy = Z[2] / cos(tx);
+    double ty = atan2(Z[0], cosy);
+    Rotx = (tx * RAD_TO_DEG);
+    Roty = (ty * RAD_TO_DEG);
+    Rotz = (tz * RAD_TO_DEG);
+  } else // roty = 90 0r 270 def
+  {
+    double k = 1;
+    if (Z[0] < 0)
+      k = -1;
+    double roty = k * 90;
+    double rotx = 0;
+    double rotz = atan2(X[2], Y[2]);
+
+    Rotx = (rotx * RAD_TO_DEG);
+    Roty = (roty * RAD_TO_DEG);
+    Rotz = (rotz * RAD_TO_DEG);
+  }
+}
+
+/**
+ * Updates the ParameterMap for NewInstrument to reflect the position of the
```

Comment:
Updates the `DetectorInfo`. is it wise to refer to the ParameterMap in any of this?